compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -884,7 +884,9 @@ impl<'a> Parser<'a> {
                 assert!(suffix.is_none());
                 let symbol = Symbol::intern(&i);
                 self.token = Token::new(token::Ident(symbol, false), ident_span);
-                let next_token = Token::new(token::Dot, dot_span);
+                // This `.` is part of a floating point literal, so it cannot
+                // be glued to anything else (e.g. to form a  `..`)
+                let next_token = (Token::new(token::Dot, dot_span), Spacing::Alone);
```

Comment:
```suggestion
                let next_token = (Token::new(token::Dot, dot_span), self.token.spacing);
```
Safe choice, you don't even need a comment.